misc/systemd/ipfs-hardened.service

```diff
@@ -0,0 +1,70 @@
+# This file will be overwritten on package upgrades, avoid customizations here.
+#
+# To make persistant changes, create file in 
+# "/etc/systemd/system/ipfs.service.d/overwrite.conf" with 
+# `systemctl edit ipfs.service`. This file will be parsed after this 
+# file has been parsed.
+#
+# To overwrite a variable, like ExecStart you have to specify it once
+# blank and a second time with a new value, like:
+# ExecStart=
+# ExecStart=/usr/bin/ipfs daemon --flag1 --flag2
+#
+# For more info about custom unit files see systemd.unit(5).
+
+# This service file enables systemd-hardening features compatible with IPFS,
+# while breaking compability with the fuse-mount function. Use this one only 
+# if you don't need the fuse-mount functionality.
+
+[Unit]
+Description=InterPlanetary File System (IPFS) daemon
+Documentation=https://docs.ipfs.io/
+After=network.target
+
+[Service]
+# hardening
+ReadWritePaths="/var/lib/ipfs/"
+NoNewPrivileges=true
+ProtectSystem=strict
+ProtectKernelTunables=true
+ProtectKernelModules=true
+ProtectKernelLogs=true
+PrivateDevices=true
+DevicePolicy=closed
```

Comment:
This is implied by PrivateDevices, right?

Reply:
Well, the other way around, `PrivateDevices` will set `DevicePolicy=closed`.

The reasoning for setting both is, some older systemd-versions might support just the simpler `DevicePolicy=closed` but doesn't yet support `PrivateDevices`.

In this case `PrivateDevices` would be ignored and `DevicePolicy=closed` not set.